### PR TITLE
Integrate webix datatable

### PR DESCRIPTION
**Pull request by Ingvord**
Proof of the concept

Peek 2024-06-15 01-57

### CODE REVIEW

File: src/app/datasets/dataset-datatable/dataset-datatable.component.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+              this.getTopParentView().refreshColumns();
+            }
+          },
+        },
+      },
+    };
+    this.webixUi.refreshColumns();
+  }
+
+  getWebixColumns(): any[] {
```
Comment (Ingvord) [inline src/app/datasets/dataset-datatable/dataset-datatable.component.ts:169]: This method shows various possible (not all though) column configuration options. See [Configuring Columns](https://docs.webix.com/datatable__columns_configuration.html) for more details

File: src/app/datasets/dataset-datatable/dataset-datatable.component.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+  ngOnDestroy(): void {
+    this.subscriptions.forEach((subscription) => subscription.unsubscribe());
+    if (this.webixUi) {
+      this.webixUi.destructor();
+    }
+  }
+
+  initializeWebixTable(): void {
+    this.webixUi = webix.ui({
```
Comment (Ingvord) [inline src/app/datasets/dataset-datatable/dataset-datatable.component.ts:95]: webix [Datatable](https://docs.webix.com/datatable__index.html) creation and configuration

**Comment by nitrosx** [comment]
Thanks for the proof-of-concept.
I'm hesitant to include another full fledged toolkit in the FE just to add the table feature.
I worry that we are going to complicate more the code and make it even more difficult to manage and maintain.

**Comment by Ingvord** [comment]
Hi Max, thanks for the feedback!

Webix has many more widgets, not just tables. I would assume to replace all
visual components with it, hence provide consistent look and feel
experience.

However,  I agree with you. This will introduce whole new abstraction layer
and may increase complexity and maintenance costs.

**Comment by sbliven** [comment]
We talked about this PR in the meeting today. A couple concerns came up.

1. **Pricing** Are all the features we would need available in webix standard (the GPL version)? IMO webix pro licenses are out of the question.
2. **Client-side filtering**. I think users would be very confused that filters are done client side and apply only to the current page of results. I think it needs to be done server-side like the facets, or else it should be omitted from the table.
3. **Maintenance**. It seems like adding webix would increase the complexity of the frontend significantly, and could increase the maintenance burden. I get that this is just a proof of concept, but while the sorting and resizing features shown here would be nice they would hardly be worth it if they make the frontend harder to maintain and develop. Do you see some "killer features" that would have a high impact on the user experience that would come with webix but not available with the current framework?

**Comment by Ingvord** [comment]
Following the discussions at SciCatCon 2024 I will repurpose this PR to demonstrate how to plugins in SciCat FE

**Pull request closed without merging**